internal/repository/rsl_test.go

```diff
@@ -239,6 +239,62 @@ func TestRecordRSLAnnotation(t *testing.T) {
 	assert.True(t, annotation.Skip)
 }
 
+func TestInvalidateLatestRSLEntry(t *testing.T) {
+	r, err := git.Init(memory.NewStorage(), memfs.New())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	repo := &Repository{r: r}
+
+	if err := rsl.InitializeNamespace(repo.r); err != nil {
+		t.Fatal(err)
+	}
+
+	// Add first RSL entry
+	refName := "refs/heads/main"
+	firstTestHash := plumbing.NewHash("1234567890abcdef")
+
+	ref := plumbing.NewHashReference(plumbing.ReferenceName("refs/heads/main"), firstTestHash)
+
+	if err := repo.r.Storer.SetReference(ref); err != nil {
+		t.Fatal(err)
+	}
+	if err := repo.RecordRSLEntryForReference(refName, false); err != nil {
+		t.Fatal(err)
+	}
+
+	// Add second RSL entry
+	secondTestHash := plumbing.NewHash("abcdef1234567890")
+
+	ref = plumbing.NewHashReference(plumbing.ReferenceName("refs/heads/main"), secondTestHash)
+
+	if err := repo.r.Storer.SetReference(ref); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := repo.RecordRSLEntryForReference(refName, false); err != nil {
+		t.Fatal(err)
+	}
+
+	// Invalidate the latest RSL entry
+	if err := repo.RollbackLatestRSLEntry(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify the latest RSL entry is now the first one
+	latestEntry, err := rsl.GetLatestEntry(repo.r)
+	if err != nil {
+		t.Fatal(err)
+	}
+	latestRefEntry, ok := latestEntry.(*rsl.ReferenceEntry)
+	if !ok {
+		t.Fatal(fmt.Errorf("invalid entry type"))
+	}
+	assert.Equal(t, refName, latestRefEntry.RefName)
+	assert.Equal(t, firstTestHash, latestRefEntry.TargetID)
```

Comment:
can you assert that latestRefEntry.ID is as expected? This test would theoretically pass with a different RSL entry that has the same contents.

Reply:
I have added the check. I have concatenated it down into `assert.Equal(t, latestGoodEntry, latestRefEntry)`. If this is not intuitive, I can roll the changes back to check all elements of the RSL entry separately.